app/Main.hs

```diff
@@ -1,7 +1,23 @@
 module Main where
 
-import ProtoRoute (someFunc)
+import Language.Haskell.Ghcid  (execStream, startGhci, stopGhci)
+import System.Directory (getCurrentDirectory)
 
 main :: IO ()
-main = someFunc
+main = putCommands
 
+putCommands :: IO ()
+putCommands = do
+    let f = \_ s -> putStrLn s
+    curDir <- getCurrentDirectory
+    (g, _) <- startGhci "ghci" (Just curDir) f
+    let execStatement s = execStream g s f
+    execStatement "import Data.Text"
+    execStatement "import Data.ProtoLens.Encoding"
+    execStatement ":load src/ProtoExports"
+    execStatement "let sr = SearchRequest {_SearchRequest'query = pack \"test\"\
+                           \, _SearchRequest'_unknownFields = ([])}"
+    -- Value of generated msg, and its serialized representation
+    execStatement "sr"
+    execStatement "encodeMessage sr"
+    stopGhci g
```

Comment:
Okay, let's actually move this chunk of code into `library` stanza. So `Main.hs` file will look like this:

```
import qualified ProtoRoute

main :: IO ()
main = ProtoRoute.main
```

This will make `executable` stanza very small (almost no dependencies). And we will put all logic into `library`. This is better for further modularity.